resources/NpmDsc/NpmDsc.psm1

```diff
@@ -1,12 +1,199 @@
 using namespace System.Collections.Generic
-#Region '.\Enum\Ensure.ps1' 0
+
+#region Functions
+function Assert-Npm {
+    # Refresh session $path value before invoking 'npm'
+    $env:Path = [System.Environment]::GetEnvironmentVariable('Path', 'Machine') + ';' + [System.Environment]::GetEnvironmentVariable('Path', 'User')
+    try {
+        Invoke-Npm -Command 'help'
+        return
+    } catch {
+        throw 'NodeJS is not installed'
+    }
+}
+
+function Invoke-Npm {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$Command
+    )
+
+    return Invoke-Expression -Command "npm $Command"
+}
+
+function Set-PackageDirectory {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$PackageDirectory
+    )
+
+    if (Test-Path -Path $PackageDirectory -PathType Container) {
+        Set-Location -Path $PackageDirectory
+    } else {
+        throw "$($PackageDirectory) does not point to a valid directory."
+    }
+}
+
+function Get-InstalledNpmPackages {
+    param (
+        [Parameter()]
+        [bool]$Global
+    )
+
+    $command = [List[string]]::new()
+    $command.Add('list')
+    $command.Add('--json')
+
+    if ($Global) {
+        $command.Add('-g')
+    }
+
+    return Invoke-Npm -Command $command
+}
+
+function Install-NpmPackage {
+    param (
+        [Parameter()]
+        [string]$PackageName,
+
+        [Parameter()]
+        [bool]$Global,
+
+        [Parameter()]
+        [string]$Arguments
+    )
+
+    $command = [List[string]]::new()
+    $command.Add('install')
+    $command.Add($PackageName)
+
+    if ($Global) {
+        $command.Add('-g')
+    }
+
+    $command.Add($Arguments)
+
+    Write-Verbose -Message "Executing 'npm $command'"
+
+    return Invoke-Npm -Command $command
+}
+
+function Uninstall-NpmPackage {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$PackageName,
+
+        [Parameter()]
+        [bool]$Global,
+
+        [Parameter()]
+        [string]$Arguments
+    )
+
+    $command = [List[string]]::new()
+    $command.Add('uninstall')
+    $command.Add($PackageName)
+
+    if ($Global) {
+        $command.Add('-g')
+    }
+
+    $command.Add($Arguments)
+
+    Write-Verbose -Message "Executing 'npm $command'"
+
+    return Invoke-Npm -Command $command
+}
+
+function GetNpmPath {
+    $npmPath = if ($IsWindows) {

```

Comment:
I think having one big if statement that assigns a value that you end up returning at the end is not really necessary. I would return the value directly rather than assigning $npmPath.

Reply:
I'll patch it up sir.